static/js/components/search/CustomSortingSelect.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Select } from 'searchkit';
+import _ from 'lodash';
+import block from 'bem-cn';
+
+import type { Event } from '../../flow/eventType';
+
+export default class CustomSortingSelect extends Select {
+
+  showDropdown(sortSelectField: string, clickEvent: Event): void {
+    clickEvent.preventDefault();
+    let event: Event = document.createEvent('MouseEvents');
+    event.initMouseEvent('mousedown', true, true, window);
+    ReactDOM.findDOMNode(sortSelectField).dispatchEvent(event);
+  }
+
+  optionText(
+    label: string,
+    showCount: number,
+    docCount: number,
+    countFormatter: Function,
+  ): string {
+    let text = label;
+
+    if (showCount && docCount !== undefined) {
+      text += ` (${countFormatter(docCount)})`;
+    }
+    return text;
+  }
+
+  render() {
+    const {
+      mod, className, items, disabled, showCount, translate, countFormatter
+    } = this.props;
+    const bemBlocks = { container: block(mod) };
+
+    return (
+      <div className={bemBlocks.container().mix(className).state({ disabled }) }>
+        <span className="label-before-selected"
+          onClick={this.showDropdown.bind(null, this.refs.sortSelectField)}>
+          Sort by:
+        </span>
+        <select onChange={this.onChange} value={this.getSelectedValue()} ref="sortSelectField">
+          {_.map(items, ({key, label, title, disabled, docCount}) => {
+            return (
+              <option key={key} value={key} disabled={disabled}>
+                {this.optionText(
+                  translate(label || title || key),
+                  showCount,
+                  docCount,
+                  countFormatter
+                )}
```

Comment:
for this `_.map` dealio here I was thinking it might make more sense to write a method called like `options` or something that takes the items and then does the rest, so that here in the JSX you're just doing `{ options(items) }`.

what do you think?